openedx/frontend-app-ora-grading
# feat: Support runtime configuration

### Description

frontend-platform supports runtime configuration since 2.5.0 (see [the PR](openedx/frontend-platform#335) that introduced it, but it requires MFE cooperation.  This implements just that: by avoiding making configuration values constant, it should now be possible to change them after initialization.

Almost all changes here relate to the `LMS_BASE_URL` setting, which in most places was treated as a constant.

For an example of a similar PR merged to another MFE, see https://github.com/openedx/frontend-app-discussions/pull/386.

### Testing

For any non-Tutor deployment, this should be a NOOP.  In other words, things should behave exactly as before.

For Tutor deployments, this is essential: without it, the MFE fails right off the bat, as the `initialize` API call falls on the wrong endpoint.  So, to test it's working, one has to:

1. Check out [this PR](https://github.com/overhangio/tutor-mfe/pull/89) to tutor-mfe and pip-install it on a virtual environment.
2. Deploy a Palm Tutor development environment (see [this branch](https://github.com/overhangio/tutor-mfe/tree/palm)).  Warning: you're going to need Docker version 23 and above for it to work!
3. Check frontend-app-ora-grading out at the branch in this PR
4. Install node dependencies with `npm ci`
5. `tutor config save --append MOUNTS=/path/to/frontend-app-ora-grading`
7. `tutor dev launch`
8. Create a new course with an ORA exercise in it.  Create a response to it, submit it, then click on "Grade available responses" as a staff user.

You should be taken to the ORA grading MFE.  There will be a couple of compilation warnings, but after closing the warning modal, your submission should be there, and the MFE should otherwise work normally.
